lib/checks/disks.go

```diff
@@ -80,55 +82,121 @@ func fioEtcdJob(filename string) *proto.CheckDisksRequest {
+	// sequential write iops required for etcd to perform effectively.
 	//
 	// The number is recommended by etcd documentation:
 	// https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/hardware.md#disks
 	//
-	EtcdMinWriteIOPS = 50
-
-	// EtcdMaxFsyncLatencyMs defines the maximum fsync latency required for
-	// etcd to perform effectively, in milliseconds.
+	// The soft threshold will generate a warning.
+	EtcdMinWriteIOPSSoft = 50
+	// EtcdMinWriteIOPSHard is the lowest number of IOPS Gravity will tolerate
+	// before generating a critical probe failure.
+	EtcdMinWriteIOPSHard = 10
+
+	// EtcdMaxFsyncLatencyMsSoft defines the soft threshold for a maximum fsync
+	// latency required for etcd to perform effectively, in milliseconds.
 	//
 	// Etcd documentation recommends 10ms for optimal performance but we're
 	// being conservative here to ensure better dev/test experience:
 	// https://github.com/etcd-io/etcd/blob/master/Documentation/faq.md#what-does-the-etcd-warning-failed-to-send-out-heartbeat-on-time-mean
 	//
-	EtcdMaxFsyncLatencyMs = 50
+	// The soft threshold will generate a warning.
+	EtcdMaxFsyncLatencyMsSoft = 50
+	// EtcdMaxFsyncLatencyMsHard is the highest fsync latency Gravity prechecks
+	// will tolerate before generating a critical probe failure.
+	EtcdMaxFsyncLatencyMsHard = 150
 
 	// testFile is the name of the disk performance test file.
 	testFile = "fio.test"
 )
+
+// getEtcdMinIOPSSoft returns the soft limit for minimum number of IOPS.
+func getEtcdMinIOPSSoft() float64 {
+	value, err := utils.GetenvInt(EtcdMinIOPSSoftEnvVar)
+	if err == nil {
+		return float64(value)
+	}
+	return EtcdMinWriteIOPSSoft
+}
+
+// getEtcdMinIOPSHard returns the hard limit for minimum number of IOPS.
+func getEtcdMinIOPSHard() float64 {
+	value, err := utils.GetenvInt(EtcdMinIOPSHardEnvVar)
+	if err == nil {
+		return float64(value)
+	}
+	return EtcdMinWriteIOPSHard
+}
+
+// getEtcdMaxLatencySoft returns the soft limit for maximum fsync latency.
+func getEtcdMaxLatencySoft() int64 {
+	value, err := utils.GetenvInt(EtcdMaxLatencySoftEnvVar)
+	if err == nil {
+		return int64(value)
+	}
+	return EtcdMaxFsyncLatencyMsSoft
+}
+
+// getEtcdMaxLatencyHard returns the hard limit for maximum fsync latency.
+func getEtcdMaxLatencyHard() int64 {
+	value, err := utils.GetenvInt(EtcdMaxLatencyHardEnvVar)
+	if err == nil {
+		return int64(value)
+	}
+	return EtcdMaxFsyncLatencyMsHard
+}
+
+const (
+	// EtcdMinIOPSSoftEnvVar is the environment variable with soft IOPS limit.
+	EtcdMinIOPSSoftEnvVar = "GRAVITY_ETCD_MIN_IOPS_SOFT"
+	// EtcdMinIOPSHardEnvVar is the environment variable with hard IOPS limit.
+	EtcdMinIOPSHardEnvVar = "GRAVITY_ETCD_MIN_IOPS_HARD"
+	// EtcdMaxLatencySoftEnvVar is the environment variable with soft fsync limit.
+	EtcdMaxLatencySoftEnvVar = "GRAVITY_ETCD_MAX_LATENCY_SOFT"
+	// EtcdMaxLatencyHardEnvVar is the environment variable with hard fsync limit.
+	EtcdMaxLatencyHardEnvVar = "GRAVITY_ETCD_MAX_LATENCY_HARD"
```

Comment:
Mentioning these overrides (and default values?) at https://gravitational.com/gravity/docs/requirements/#etcd-disk wouldn't go awry.

Reply:
Yeah, I was actually going to mention these environment variables in the [corresponding docs section](https://gravitational.com/gravity/docs/installation/#environment-variables) as well - thanks for reminding.

Reply:
Added here: https://github.com/gravitational/gravity/pull/1859/commits/6190f1a8baf6dd59068e4459b449d1e7ec716083.